### ISSUE TITLE
Does the --superheat option work?

### ISSUE THREAD

**Issue by grosu** [kframework/k-legacy#2034]
I'm working on revising lambda++ now, and in lesson 2 we should have 4 different solutions for free-variable-capture.lambda, when kompiled with option --superheat strict.  In old unparsing, they used to be:

```
Solution 1:
V:K -->
a ~> (HOLE ((lambda x . (lambda y . x) y) z))

Solution 2:
V:K -->
y ~> closure ( .Map , x , lambda y . x ) HOLE ~> HOLE z ~> (a HOLE)

Solution 3:
V:K -->
y ~> lambda x . (lambda y . x) HOLE ~> HOLE z ~> (a HOLE)

Solution 4:
V:K -->
z ~> (lambda x . (lambda y . x) y) HOLE ~> (a HOLE)
```

However, I can now only see one solution, no matter what I try:

```
Solution 1:
V:K -->
z ~> #freezer1 ( ( lambda x . lambda y . x ) y ) ~> #freezer1 ( a )
```

So, does the --superheat option work?  It seems to work in imp++, where it finds the 5 solutions of div.imp, as it should.

**Comment by cos** [comment]
It is a bit curious it works for div. @andreistefanescu, do you have any idea?

The strategy-based heating/cooling, which can do superheat if given the right strategy, is only activated if importing the STRATEGY module, which is not currently the default.

**Comment by grosu** [comment]
Actually, I did not really check div.imp, I only looked at the config.xml files and the div.imp.out file, and assumed that @laurayuwen just ran ktest when she revised imp++ and was done with it:

https://github.com/kframework/k/blob/3.6Base/k-distribution/tutorial/1_k/4_imp%2B%2B/lesson_3/tests/div.imp.out
https://github.com/kframework/k/blob/3.6Base/k-distribution/tutorial/1_k/4_imp%2B%2B/lesson_3/tests/config.xml
https://github.com/kframework/k/blob/3.6Base/k-distribution/tutorial/1_k/4_imp%2B%2B/tests/config.xml
https://github.com/kframework/k/blob/3.6Base/k-distribution/tutorial/1_k/tests/config.xml

I'll check if it actually works or not later when I get a chance, but probably you are right and we have some other kind of bug in how ktest checks the regexps in .out files.

**Comment by laurayuwen** [comment]
div has been excluded from lesson 1, as later lessons use test config from previous lessons, so you don't see that directly in current lesson. When I worked on it, I couldn't get multiple results either, so left it for later.

**Comment by daejunpark** [comment]
At that time I worked on KORE-translatoin of lambda++, `--superheat` wan't supported, so `free-variable-capture` was excluded from lesson_2:
https://github.com/kframework/k/blob/e6a13a83d395aee3bd330609d061b27821674fb4/k-distribution/tutorial/1_k/3_lambda%2B%2B/lesson_2/tests-kore/config.xml#L5-L8
As such, lesson_4 had several excludes as well:
https://github.com/kframework/k/blob/e6a13a83d395aee3bd330609d061b27821674fb4/k-distribution/tutorial/1_k/3_lambda%2B%2B/lesson_4/tests-kore/config.xml#L16-L18

Regarding imp++, as @laurayuwen said, she excluded `div` from lesson_1 of imp++:
https://github.com/kframework/k/blob/e6a13a83d395aee3bd330609d061b27821674fb4/k-distribution/tutorial/1_k/4_imp%2B%2B/lesson_1/tests/config.xml#L8
Also, lesson_6 has more excludes:
https://github.com/kframework/k/blob/e6a13a83d395aee3bd330609d061b27821674fb4/k-distribution/tutorial/1_k/4_imp%2B%2B/lesson_6/tests/config.xml#L7

**Comment by andreistefanescu** [comment]
Fixed in #2038 .
